ttnn/cpp/ttnn/operations/full_like/device/full_like_factory.cpp

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+#include "common/constants.hpp"
+#include "full_like_device_operation.hpp"
+#include "host_api.hpp"
+#include "impl/buffers/circular_buffer_types.hpp"
+#include "ttnn/tensor/tensor_utils.hpp"
+#include "ttnn/tensor/types.hpp"
+#include "tt_metal/common/work_split.hpp"
+
+namespace ttnn::operations::full_like {
+
+using namespace tt;
+using namespace tt::tt_metal;
+using namespace tt::constants;
+
+
+union datatype {
+    uint32_t u32;
+    float f32;
+} u;
+
+FullLikeOperation::ProgramFactory::cached_program_t FullLikeOperation::ProgramFactory::create(
+    const operation_attributes_t &operation_attributes,
+    const tensor_args_t &tensor_args,
+    tensor_return_value_t &output) {
+    auto input = tensor_args.input;
+    auto fill_value = operation_attributes.fill_value;
+    if (std::holds_alternative<int>(fill_value)) {
+        u.u32 = std::get<int>(fill_value);
+    } else if (std::holds_alternative<float>(fill_value)) {
+        u.f32 = std::get<float>(fill_value);
+    }
+    DataType dtype{operation_attributes.dtype};
+    Layout layout{operation_attributes.layout};
+    Device *device = input.device();
+    MemoryConfig memory_config{operation_attributes.memory_config};
+
+    auto num_tiles = input.volume() / TILE_HW;
+
+    Program program{};
+
+    tt::DataFormat data_format = tt::tt_metal::datatype_to_dataformat_converter(dtype);
+    uint32_t single_tile_size = tt::tt_metal::detail::TileSize(data_format);
+
+    const auto compute_with_storage_grid_size = device->compute_with_storage_grid_size();
+    const uint32_t num_cores_x = compute_with_storage_grid_size.x;
+    const uint32_t num_cores_y = compute_with_storage_grid_size.y;
+
+    auto [num_cores, all_cores, core_group_1, core_group_2, num_tiles_per_core_group_1, num_tiles_per_core_group_2] =
+        tt_metal::split_work_to_cores(compute_with_storage_grid_size, num_tiles);
+
+    constexpr CB cb_fill_value = CB::c_intermed0;
```

Comment:
Rename cb_fill_value to cb_fill_value_id or cb_fill_value_index

Reply:
Resolved